.github/workflows/demo-app-build-and-deploy.yml

```diff
@@ -0,0 +1,82 @@
+name: Build and deploy demo app
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version of the app (e.g. v1.2.3)'
+        required: true
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      TF_VAR_DOCKER_USERNAME: ${{ secrets.TF_VAR_DOCKER_USERNAME }}
+      TF_VAR_DOCKER_PASSWORD: ${{ secrets.TF_VAR_DOCKER_PASSWORD }}
+      TF_VAR_DOCKER_EMAIL: ${{ secrets.TF_VAR_DOCKER_EMAIL }}
+      TF_VAR_DOCKER_IMAGE: "${{ secrets.TF_VAR_DOCKER_IMAGE }}:${{ github.event.inputs.version }}"
+      TF_VAR_KEYCLOAK_AUTHORITY: ${{ secrets.TF_VAR_KEYCLOAK_AUTHORITY }}
+      TF_VAR_APP_1_KEYCLOAK_CLIENT_ID: ${{ secrets.TF_VAR_APP_1_KEYCLOAK_CLIENT_ID  }}
+      TF_VAR_APP_1_KEYCLOAK_CLIENT_SECRET: ${{  secrets.TF_VAR_APP_1_KEYCLOAK_CLIENT_SECRET }}
+      TF_VAR_APP_2_KEYCLOAK_CLIENT_ID : ${{  secrets.TF_VAR_APP_2_KEYCLOAK_CLIENT_ID }}
+      TF_VAR_APP_2_KEYCLOAK_CLIENT_SECRET : ${{  secrets.TF_VAR_APP_2_KEYCLOAK_CLIENT_SECRET }}
+
+    steps:
+      - uses: actions/checkout@v2
+        name: checkout
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Login to docker repository
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.TF_VAR_DOCKER_USERNAME }}
+          password: ${{ secrets.TF_VAR_DOCKER_PASSWORD }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build Docker Image - Demo App
+        uses: docker/build-push-action@v3
+        with:
+          file: demo-app/.docker/Dockerfile.prod
+          context: ./demo-app
+          push: true
+          tags: "${{ secrets.TF_VAR_DOCKER_IMAGE }}:${{ github.event.inputs.version }}"
+
+      - name: Create Kube Config
+        run: |
+          echo "${{secrets.KUBE_CONFIG}}" > ./terraform/demo-app/kubeconfig
+      - name: set kube-config
+        uses: azure/k8s-set-context@v2
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.KUBE_CONFIG }}
+          context: ${{ secrets.KUBE_CONTEXT }}
+
+      - name: terraform destroy
```

Comment:
Do we need this `destroy` action? Seeing as we're passing along the Docker image version, it *should* redeploy it (hopefully)